code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/TimeUtils.kt

```diff
@@ -36,45 +39,66 @@ object TimeUtils {
-     * TimeZone format used is ISO8601 using XXX formatting letters. ex: 9th July 2020 PDT will be formatted as 2020-07-09T15:09:18-07:00.
+     * Gets the ISO 8601 formatted with UTC(Z) time zone, millisecond precision date `String` for the
+     * provided date using the UTC +0 time zone.
+     * Date pattern used is [DatePattern.ISO8601_TIMEZONE_ISO8601_UTCZ_PRECISION_MILLISECOND]
+     * which has timezone ISO 8601 'Z' char terminator, which gives a timezone of 'Z'.
+     * Ex: Wed Nov 30 11:53:09.497 GMT-07:00 2022 -> 2022-11-30T18:53:09.497Z (notice the hour shift because the date must be evaluated from UTC +0)
+     *
+     * Note that ISO 8601 requires date strings terminating with the char 'Z' to be in the time zone UTC +0 (no offset).
+     *
      * AMSDK-10273 :: ExEdge requires time zone offset formatted in form [+-]HH:MM.
      *
-     * @return Iso8601 formatted date [String]
+     * @param date the [Date] to apply the formatting to; defaults to the current [Date]
+     * @return date [String] formatted as ISO 8601 with UTC(Z) time zone, millisecond precision
      */
     @JvmStatic
-    fun getIso8601DateTimeZoneISO8601(): String? {
-        return getIso8601Date(Date(), ISO8601_DATE_FORMATTER_TIMEZONE_UTC, TimeZone.getTimeZone("GMT"))
+    @JvmOverloads
+    fun getISO8601UTCDateWithMilliseconds(date: Date? = Date()): String? {
+        return getFormattedDate(date, DatePattern.ISO8601_TIMEZONE_ISO8601_UTCZ_PRECISION_MILLISECOND)
     }
 
     /**
-     * Gets the the ISO 8601 formatted date `String` for the passed in date - "yyyy-MM-dd'T'HH:mm:ssZZZ"
+     * Gets the formatted date `String` for the passed in date; if no date passed in, uses current `Date()`
      *
-     * @param date the [Date] to generate the [String] for
-     * @return ISO 8601 formatted date [String]
+     * @param date the [Date] to apply the formatting to; defaults to the current [Date]
+     * @param datePattern the [DatePattern] to use to format the date [String]
+     * @param timeZone the [TimeZone] to evaluate the formatted date from; defaults to device time zone if not specified
+     * @return the formatted date [String]
      */
-    @JvmStatic
-    @JvmOverloads
-    fun getIso8601Date(date: Date?, format: String?, timeZone: TimeZone? = null): String? {
+    private fun getFormattedDate(date: Date?, datePattern: DatePattern, timeZone: TimeZone? = null): String? {
         // AMSDK-8374 -
         // we should explicitly ignore the device's locale when formatting an ISO 8601 timestamp
         val posixLocale = Locale(Locale.US.language, Locale.US.country, "POSIX")
-        val iso8601Format: DateFormat = SimpleDateFormat(format, posixLocale)
-        if (timeZone != null) {
-            iso8601Format.timeZone = timeZone
+        val dateFormat: DateFormat = SimpleDateFormat(datePattern.pattern, posixLocale)
+
+        // Handle special cases/requirements based on DatePattern provided
+        when (datePattern) {
```

Comment:
Let us remove this additional logic and make it the responsibility of caller to set the correct timezone.